src/documents.jl

```diff
@@ -688,6 +688,63 @@ function lt_remotepair(r1::RemoteRepository, r2::RemoteRepository)
     return length(r1.root) > length(r2.root)
 end
 
+src_to_uuid = nothing
+uuid_to_version = nothing
+
+function build_dep_info_dicts()
+    global src_to_uuid
+    global uuid_to_version
+    src_to_uuid = Dict()
+    uuid_to_version = Dict()
+
+    for (uuid, dep) in pairs(Pkg.dependencies())
+        src_to_uuid[dep.source] = uuid
+        uuid_to_version[uuid] = dep.version
+    end
+end
```

Comment:
Globals, let's try to avoid those :grimacing: Could we move these to being fields of `doc.internal`? We'd be re-populating them if you call `makedocs` multiple times in a REPL session, but that's fine I think?